compiler/resolution/visibleFunctions.cpp

```diff
@@ -482,20 +477,61 @@ static void lookAtTypeFirst(const char* name, CallExpr* call, BlockStmt* block,
   }
   Type* t = typeActual->getValType();
 
-  BlockStmt* typeScope = getVisibilityScope(t->symbol->defPoint);
-  // When searching the type scope, don't follow private uses and imports if we
-  // aren't already in the scope where the type is defined.
-  bool inScopeJump = typeScope != block;
+  lookAtTypeFirstHelper(name, call, block, t, visInfo, visited, visibleFns);
 
-  // Look at own methods
-  getVisibleMethodsImpl(name, call, typeScope, visInfo, visited, visibleFns,
-                        inScopeJump);
+  Type* canonT = canonicalClassType(t);
```

Comment:
I'm uncertain about checking the canonical class type. It seems we would want to check `forwarding` things more generally, but if there's a problem with that, there's also probably a problem with `owned` -> `borrowed`. I suppose that the difference is that we are willing to implicitly convert from `owned` to `borrowed`. Anyway is there an issue or something about this?

Reply:
We talked about this in Cray/chapel-private#1731.  Specifically you said in https://github.com/Cray/chapel-private/issues/1731#issuecomment-836762093 that you'd be fine with it

Reply:
Ah, thanks for helping me remember. In that issue we were talking about `==` which returns `bool`. The case I am concerned about is an operator that returns something of the same type, for example `+`. I am worried that the type of the result of `+` might be surprising. I think that we could extend the reasoning you have here to all `forwarded` operators that return `bool` (or perhaps to such operators for types that the type can implicitly convert to).  Anyway I think it's a bit moot in practice for classes anyway since presumably `proc +` would have to define a particular ownership style for the return (rather than returning `borrowed SomeClass`. So, all that is to say, I think we should make an issue asking:
 1. If this behavior should be extended to other `forwarding` or implicit conversion situations
 2. If it does something reasonable specifically for operators returning something other than `bool` e.g. `+`

Reply:
I opened #17717, does it accurately summarize your concerns?

Reply:
Yes, it's great, thanks for creating it.